src/empirical.jl

```diff
@@ -47,7 +50,12 @@ evaluate CDF values on other samples.
 `extrema`, `minimum`, and `maximum` are supported to for obtaining the range over which
 function is inside the interval ``(0,1)``; the function is defined for the whole real line.
 """
-ecdf(X::RealVector{T}) where T<:Real = ECDF(sort(X))
+function ecdf(X::RealVector{T}, w::AbstractWeights{W} = weights(ones(length(X)))) where {T, W <: Real}
```

Comment:
Instead of allocating a `ones` vector, could you use `Float64[]` when no weights are provided? That way you don't need to allocate a vector for the unweighted case. Then functions should check whether `isempty(X.weights)` to decide what to do.

Reply:
Alright, I took a crack at it. Let me know what you think.